src/test/unit_tests/test_nc_health.js

```diff
@@ -645,23 +667,27 @@ mocha.describe('nsfs nc health', function() {
  */
 function assert_config_dir_status(health_status, expected_config_dir) {
     const actual_config_dir_health = health_status.checks.config_directory_status;
+    assert.strictEqual(actual_config_dir_health.error, expected_config_dir.error);
+    assert.deepStrictEqual(actual_config_dir_health.blocked_hosts, expected_config_dir.blocked_hosts);
     assert.strictEqual(actual_config_dir_health.phase, expected_config_dir.phase);
     assert.strictEqual(actual_config_dir_health.config_dir_version, expected_config_dir.config_dir_version);
     assert.strictEqual(actual_config_dir_health.upgrade_package_version, expected_config_dir.upgrade_package_version);
-    const expected_upgrade_status = expected_config_dir.in_progress_upgrade || expected_config_dir.upgrade_history?.last_failure;
-    assert_upgrade_status(actual_config_dir_health.upgrade_status, expected_upgrade_status);
+    assert_upgrade_status(actual_config_dir_health.upgrade_status, expected_config_dir);
 }
 
 /**
  * assert_upgrade_status asserts there the actual and expected upgrade_status
  * @param {Object} actual_upgrade_status 
- * @param {Object} [expected_upgrade_status]
+ * @param {Object} expected_upgrade_status
  */
-function assert_upgrade_status(actual_upgrade_status, expected_upgrade_status = default_mock_upgrade_status.message) {
-    const actual_upgrade_status_res = actual_upgrade_status.in_progress_upgrade ||
-        actual_upgrade_status.last_failure ||
-        actual_upgrade_status.message;
-    assert.deepStrictEqual(actual_upgrade_status_res, expected_upgrade_status);
+function assert_upgrade_status(actual_upgrade_status, expected_upgrade_status) {
+    const actual_upgrade_status_res = actual_upgrade_status?.in_progress_upgrade ||
+        actual_upgrade_status?.last_failure ||
+        actual_upgrade_status?.message;
+    const expected_upgrade_status_res = expected_upgrade_status?.in_progress_upgrade ||
+        expected_upgrade_status?.upgrade_history?.last_failure ||
+        expected_upgrade_status?.upgrade_status?.message;
+    assert.deepStrictEqual(actual_upgrade_status_res, expected_upgrade_status_res);
```

Comment:
What does it mean if both `actual_upgrade_status_res` and `expected_upgrade_status_res` are `undefined`?

Reply:
see the error - config directory data is missing, must upgrade config directory